runtime/src/native_loader.rs

```diff
@@ -134,23 +134,29 @@ impl NativeLoader {
     pub fn process_instruction(
         &self,
         program_id: &Pubkey,
-        keyed_accounts: &[KeyedAccount],
         instruction_data: &[u8],
-        invoke_context: &dyn InvokeContext,
+        invoke_context: &mut dyn InvokeContext,
     ) -> Result<(), InstructionError> {
-        let mut keyed_accounts_iter = keyed_accounts.iter();
-        let program = next_keyed_account(&mut keyed_accounts_iter)?;
-        if native_loader::id() != *program_id {
-            error!("Program id mismatch");
-            return Err(InstructionError::IncorrectProgramId);
-        }
-        if program.owner()? != *program_id {
-            error!("Executable account now owned by loader");
-            return Err(InstructionError::IncorrectProgramId);
-        }
+        let (program_id, name_vec) = {
+            let keyed_accounts = invoke_context.get_keyed_accounts();
+            let program = keyed_account_at_index(keyed_accounts, 0)?;
+            if native_loader::id() != *program_id {
+                error!("Program id mismatch");
+                return Err(InstructionError::IncorrectProgramId);
+            }
+            if program.owner()? != *program_id {
+                error!("Executable account now owned by loader");
+                return Err(InstructionError::IncorrectProgramId);
+            }
+            // TODO: Remove these two copies (* deref is also a copy)
```

Comment:
I really dislike introducing more copy / clone calls, so I marked this.
But, the comment in the code can probably be removed.